pkg/sensors/program/loader_windows.go

```diff
@@ -0,0 +1,480 @@
+	return func(_ *ebpf.Collection, _ *ebpf.CollectionSpec,
+		prog *ebpf.Program, spec *ebpf.ProgramSpec) (unloader.Unloader, error) {
+
+		unloader := unloader.ChainUnloader{
+			unloader.ProgUnloader{
+				Prog: prog,
+			},
+		}
+
+		for idx := range syms {
+			un, err := kprobeAttach(load, prog, spec, syms[idx], bpfDir, fmt.Sprintf("%d_%s", idx, syms[idx]))
+			if err != nil {
+				return nil, err
+			}
+
+			unloader = append(unloader, un)
+		}
+		return unloader, nil
+	}
+}
+
+func LoadKprobeProgramAttachMany(bpfDir string, load *Program, syms []string, verbose int) error {
+	opts := &LoadOpts{
+		Attach: KprobeAttachMany(load, syms, bpfDir),
+	}
+	return loadProgram(bpfDir, load, opts, verbose)
+}
+
+func LoadUprobeProgram(bpfDir string, load *Program, verbose int) error {
+	opts := &LoadOpts{
+		Attach: UprobeAttach(load),
+	}
+	return loadProgram(bpfDir, load, opts, verbose)
+}
+
+func LoadMultiKprobeProgram(bpfDir string, load *Program, verbose int) error {
+	opts := &LoadOpts{
+		Attach: MultiKprobeAttach(load, bpfDir),
+		Open:   KprobeOpen(load),
+	}
+	return loadProgram(bpfDir, load, opts, verbose)
+}
+
+func LoadFmodRetProgram(bpfDir string, load *Program, progName string, verbose int) error {
+	return fmt.Errorf("not supported on windows")
+}
+
+func LoadTracingProgram(bpfDir string, load *Program, verbose int) error {
+	opts := &LoadOpts{
+		Attach: TracingAttach(),
+	}
+	return loadProgram(bpfDir, load, opts, verbose)
+}
+
+func LoadLSMProgram(bpfDir string, load *Program, verbose int) error {
+	opts := &LoadOpts{
+		Attach: LSMAttach(),
+		Open:   LSMOpen(load),
+	}
+	return loadProgram(bpfDir, load, opts, verbose)
+}
+
+func LoadLSMProgramSimple(bpfDir string, load *Program, verbose int) error {
+	opts := &LoadOpts{
+		Attach: LSMAttach(),
+	}
+	return loadProgram(bpfDir, load, opts, verbose)
+}
+
+func LoadMultiUprobeProgram(bpfDir string, load *Program, verbose int) error {
+	opts := &LoadOpts{
+		Attach: MultiUprobeAttach(load),
+	}
+	return loadProgram(bpfDir, load, opts, verbose)
+}
+
+// MissingConstantsError is returned by [rewriteConstants].
+type MissingConstantsError struct {
+	// The constants missing from .rodata.
+	Constants []string
+}
+
+func (m *MissingConstantsError) Error() string {
+	return fmt.Sprintf("some constants are missing from .rodata: %s", strings.Join(m.Constants, ", "))
+}
+
+func doLoadProgram(
+	bpfDir string,
+	load *Program,
+	loadOpts *LoadOpts,
+	verbose int,
+) (*LoadedCollection, error) {
+
+	coll, err := ebpf.LoadCollection(load.Name)
+	if err != nil {
+		logger.GetLogger().WithError(err).WithField("Error ", err.Error()).Warn(" Failed to load Native Windows Collection ")
+		return nil, err
+	}
+	bpf.SetExecCollection(coll)
+
+	collMaps := map[ebpf.MapID]*ebpf.Map{}
+	// we need a mapping by ID
+	for _, m := range coll.Maps {
+
+		info, err := m.Info()
+		if err != nil {
+			logger.GetLogger().WithError(err).WithField("map", m.String()).Warn("failed to retrieve BPF map info")
+			continue
```

Comment:
linux code does `break` in here, does m.Info() work on windows? if not we could remove this completely for now